EKS ci for latest is unable to create a cluster

### Bug Description

Once we switched to 1.31 k8s version, the create cluster step fails with the logs below. I think we need to update the Ubuntu version used.

### To Reproduce

Rerun CI for latest/edge CKF.

### Environment

https://github.com/canonical/bundle-kubeflow/actions/runs/12877838652/job/35902921789#step:10:18

### Relevant Log Output

```shell
2025-01-21 00:37:42 [ℹ]  eksctl version 0.201.0
2025-01-21 00:37:42 [ℹ]  using region eu-central-1
2025-01-21 00:37:42 [ℹ]  subnets for eu-central-1a - public:192.168.0.0/19 private:192.168.64.0/19
2025-01-21 00:37:42 [ℹ]  subnets for eu-central-1b - public:192.168.32.0/19 private:192.168.96.0/19
Error: unable to determine AMI to use: unable to determine AMI for region eu-central-1, version 1.31, instance type t2.2xlarge and image family Ubuntu2004
Error: Process completed with exit code 1.
```

### Additional Context

_No response_

--- a/.github/cluster.yaml
+++ b/.github/cluster.yaml
@@ -14,7 +14,7 @@ kind: ClusterConfig
 kubernetesNetworkConfig:
   ipFamily: IPv4
 managedNodeGroups:
-- amiFamily: Ubuntu2004
+- amiFamily: Ubuntu2204
   iam:
     withAddonPolicies:
       ebs: true

